tests/test_pylock.py

```diff
@@ -0,0 +1,472 @@
+from __future__ import annotations
+
+import sys
+from datetime import datetime
+from pathlib import Path
+from textwrap import dedent
+from typing import Any
+
+import pytest
+import tomli_w
+
+from packaging.markers import Marker
+from packaging.pylock import (
+    Package,
+    PackageDirectory,
+    PackageVcs,
+    PackageWheel,
+    Pylock,
+    PylockRequiredKeyError,
+    PylockUnsupportedVersionError,
+    PylockValidationError,
+    is_valid_pylock_path,
+)
+from packaging.specifiers import SpecifierSet
+from packaging.version import Version
+
+if sys.version_info >= (3, 11):
+    import tomllib
+else:
+    import tomli as tomllib
+
+
+@pytest.mark.parametrize(
+    "file_name,valid",
+    [
+        ("pylock.toml", True),
+        ("pylock.spam.toml", True),
+        ("pylock.json", False),
+        ("pylock..toml", False),
+    ],
+)
+def test_pylock_file_name(file_name: str, valid: bool) -> None:
+    assert is_valid_pylock_path(Path(file_name)) is valid
+
+
+# This is the PEP 751 example, with the following differences:
+# - a minor modification to the 'environments' field to use double quotes
+#   instead of single quotes, since that is what 'packaging' does when
+#   serializing markers;
+# - added an index field, which was not demonstrated in the PEP 751 example.
+
+PEP751_EXAMPLE = dedent(
+    """\
+    lock-version = '1.0'
```

Comment:
Would it make sense to put this in its own file in a subdirectory like `metadata` and `musllinux`?